lib/util/stream_utils.js

```diff
@@ -33,10 +34,16 @@ shaka.util.StreamUtils = class {
    * @param {!Array.<string>} preferredAudioCodecs
    * @param {number} preferredAudioChannelCount
    * @param {!Array.<string>} preferredDecodingAttributes
+   * @param {shaka.config.CodecSwitchingStrategy} codecSwitchingStrategy
    */
   static chooseCodecsAndFilterManifest(manifest, preferredVideoCodecs,
       preferredAudioCodecs, preferredAudioChannelCount,
-      preferredDecodingAttributes) {
+      preferredDecodingAttributes, codecSwitchingStrategy) {
+    // If enforcing a codec switch strategy skip filtration
+    if (codecSwitchingStrategy !==
```

Comment:
This PR is just providing the ability to enable codec switching. I am currently working on the second part of this which is the logic which is implementing the logic that @sky-hugolima is describing. https://github.com/shaka-project/shaka-player/pull/4766#issuecomment-1589101381

Reply:
OK, so I hope you will upload the changes to this PR. (I'd prefer all of this changes in the same PR)

Reply:
I have added the codec selection logic, if this is what you had in mind I will perform more testing and I will write the units for those methods.

Reply:
The truth is that it does not fit with what I had in mind. We want the filtering to be done by shaka itself, by default (if someone wants to customize it, it can be made customizable). So the DISABLED option should not exist. The default option should be SMOOTH except on devices we know where it doesn't work that default should be RELOAD. In your commit the option of being able to perform external filtering is added, but there is no logic in Shaka.

Reply:
The filtering logic lives inside shaka within the methods by default `filterAudioTracksForCodecSwitching_` and `filterVideoTracksForCodecSwitching_` as per https://github.com/shaka-project/shaka-player/pull/4766#discussion_r1042049562, the comparator callback is for customisation only (unless I am not understanding correctly what you are saying). 

With regards to being enabled by default, some LG devices that I have tested on _support_ `changeType` but don't seem to handle it gracefully. Should we maintain a matrix of known supported device/OS cut offs?

Reply:
Currently we already have things in the configuration that change depending on the device/browser, for example https://github.com/shaka-project/shaka-player/blob/main/lib/util/player_configuration.js#L222 So I think in this case you can do the same. If in the case of WebOS changeType does not work we must put RELOAD by default.

Reply:
@dave-nicholas I have shared to you on slack (video-dev) some changes about the idea of codec swithcing, I hope they help